partner_event/models/event_registration.py

```diff
@@ -20,25 +26,24 @@ def _prepare_partner(self, vals):
 
     @api.model
     def create(self, vals):
-        if not vals.get('partner_id') and vals.get('email'):
-            partner_model = self.env['res.partner']
-            event_model = self.env['event.event']
-            partner_id = False
+        if not vals.get('attendee_partner_id') and vals.get('email'):
+            Partner = self.env['res.partner']
+            Event = self.env['event.event']
+            attendee_partner_id = False
             # Look for a partner with that email
             email = vals.get('email').replace('%', '').replace('_', '\\_')
-            partner = partner_model.search(
-                [('email', '=ilike', email)], limit=1)
-            event = event_model.browse(vals['event_id'])
-            if partner:
-                partner_id = partner.id
-                vals['name'] = vals.get('name') or partner.name
-                vals['phone'] = vals.get('phone') or partner.phone
+            attendee_partner = Partner.search([
+                ('email', '=ilike', email)
+            ], limit=1)
+            event = Event.browse(vals['event_id'])
+            if attendee_partner:
+                vals['name'] = vals.get('name') or attendee_partner.name
```

Comment:
Maybe force here the attendee_partner name?

Reply:
I have doubts... If in my second register with the same email I don't know what to do.
Do we want the new name or the save name in partner record? ping @rafaelbn